lib/BackendV3.js

```diff
@@ -14,50 +14,48 @@ class BackendV3 {
   constructor(config) {
     this.config = config;
     this.nodeisSyncing = true;
-    this.runCrawlers();
   }
+
   async runCrawlers() {
-    let api;
-    api = await this.getPolkadotAPI();
+    const pool = await this.getPool();
+
+    let api = await this.getPolkadotAPI();
     while (!api) {
       await wait(10000);
       api = await this.getPolkadotAPI();
     }
-    const pool = await this.getPool();
-
-    console.log(`[PolkaStats backend v3] - \x1b[32m${JSON.stringify(this.config, null, 2)}\x1b[0m`);
 
     console.log(`[PolkaStats backend v3] - \x1b[32mRunning crawlers\x1b[0m`);
-    // Listen to new blocks and system events and add them to database
-    console.log(`[PolkaStats backend v3] - \x1b[32mStarting block listener...\x1b[0m`);
-    blockListener(api, pool);
-    // Get rewards
-    console.log(`[PolkaStats backend v3] - \x1b[32mStarting rewards crawler...\x1b[0m`);
-    rewards(api, pool);
-    // Fill the gaps in block and event tables
-    setTimeout(() => {
-      console.log(`[PolkaStats backend v3] - \x1b[32mStarting block harvester...\x1b[0m`);
-      blockHarvester(api, pool, this.config);
-    }, this.config.BLOCK_HARVESTER_POLLING_TIME);
-    // Get information about active accounts
-    setTimeout(() => {
-      console.log(`[PolkaStats backend v3] - \x1b[32mStarting active accounts crawler...\x1b[0m`);
-      activeAccounts(api, pool, this.config);
-    }, this.config.ACTIVE_ACCOUNTS_POLLING_TIME);
-    // Get offline phragmen output
-    setInterval(
-      () => phragmen(api, pool, this.config.phragmen),
-      this.config.PHRAGMEN_POLLING_TIME,
-    );
-    // Get staking info
-    setTimeout(() => {
-      console.log(`[PolkaStats backend v3] - \x1b[32mStarting staking crawler...\x1b[0m`);
-      staking(api, pool);
-    }, this.config.STAKING_POLLING_TIME);
+
+    if (this.config.blockListener) {
+      blockListener(api, pool, this.config.blockListener);
+    }
+
+    if (this.config.rewards) {
+      rewards(api, pool, this.config.rewards);
+    }
+
+    if (this.config.blockHarvester) {
+      blockHarvester(api, pool, this.config.blockHarvester);
+    }
+
+    if (this.config.accounts) {
+      activeAccounts(api, pool, this.config.accounts);
+    }
+
+    if (this.config.staking) {
+      staking(api, pool, this.config.staking);
+    }
+
+    if (this.config.phragmen) {
+      phragmen(api, pool, this.config.phragmen);
+    }
```

Comment:
This one looks definitely much cleaner and readable than the old version.

Just wondering if there is a bit less repetitive way to put it, but I think this is the way to go

Reply:
We could store all the crawlers in an array and loop over it

Reply:
Let's do that :dancer:

Reply:
Looks great! Good idea, I think we can use and array for the crawlers and do a foreach to start them based on conf.